Modified ``__sub__()`` method doesn't work if there is already a hole present

See slightly modified example from #198 below, which generates a hole in the centre of the rectangle:

```python
from typing import Union, List
from sectionproperties.pre.library.concrete_sections import primitive_sections
from sectionproperties.pre.pre import Material
from sectionproperties.analysis.section import Section
import sectionproperties.pre.pre as pre
import sectionproperties.pre.geometry as geometry

def concrete_rectangular_hollow_section(
    b: float,
    d: float,
    t: float,
    dia: float,
    n_bar: int,
    n_circle: int,
    cover: float,
    area: float = None,
    conc_mat: pre.Material = pre.DEFAULT_MATERIAL,
    steel_mat: pre.Material = pre.DEFAULT_MATERIAL,
) -> geometry.CompoundGeometry:

    if n_bar < 2:
        raise ValueError("Please provide 2 or more steel reinforcing bars.")

    geom = primitive_sections.rectangular_section(b=b, d=d, material=conc_mat)
    geom = geom - primitive_sections.rectangular_section(b=b-2*t, d=d-2*t).align_center(align_to=geom)
    geom.plot_geometry()
    print(geom.holes)

    x_i = cover + dia / 2
    spacing = (b - 2 * cover - dia) / (n_bar - 1)

    for i in range(n_bar):
        if area:
            bar = primitive_sections.circular_section_by_area(
                area=area, n=n_circle, material=steel_mat
            )
        else:
            bar = primitive_sections.circular_section(
                d=dia, n=n_circle, material=steel_mat
            )

        bar = bar.shift_section(x_offset=x_i + spacing * i, y_offset=cover + dia / 2)

        # make hole for bar
        geom = geom - bar

        # add bar
        geom = geom + bar

    print(geom.holes)

    return geom


concrete = Material(
    name='Concrete', elastic_modulus=30.1e3, poissons_ratio=0.2, yield_strength=32,
    density=2.4e-6, color='lightgrey'
)
steel = Material(
    name='Steel', elastic_modulus=200e3, poissons_ratio=0.3, yield_strength=500,
    density=7.85e-6, color='grey'
)

geometry = concrete_rectangular_hollow_section(
    b=600, d=600, t=100, dia=20, n_bar=5, n_circle=24, cover=30, conc_mat=concrete, steel_mat=steel
)
geometry.create_mesh(mesh_sizes=[500])
geometry.plot_geometry()

sec = Section(geometry)
sec.plot_mesh()
```

The ``__sub__()`` method now adds holes at the location of all reinforcement.

The first ``print(geom.holes)`` results in: ``[(300.0, 300.0)]``
The second ``print(geom.holes)`` results in: ``[(300.0, 300.0), (40.0, 41.2940952255125), (170.0, 41.2940952255125), (300.0, 41.2940952255125), (430.0, 41.2940952255125), (560.0, 41.2940952255125)]``

As a result, the section cannot be created as you get the following error:
``AssertionError: Number of materials (6), should match the number of regions (1).``

Hopefully a quick an easy fix?

--- a/sectionproperties/pre/geometry.py
+++ b/sectionproperties/pre/geometry.py
@@ -1058,7 +1058,6 @@ class Geometry:
         """
         Perform difference on Geometry objects with the - operator
         """
-        material = pre.DEFAULT_MATERIAL
         if isinstance(self, CompoundGeometry):
             subs_geom_acc = []
             for geom in self.geoms:
@@ -1717,35 +1716,32 @@ class CompoundGeometry(Geometry):
                 self.facets.append([i_pnt_idx, j_pnt_idx])
 
             # add holes
+            existing_geom_holes = []
             for hole in geom.holes:
-                self.holes.append(tuple(hole))
+                existing_geom_holes.append(tuple(hole))
 
             # add control points
             for control_point in geom.control_points:
                 self.control_points.append(tuple(control_point))
 
-        # Check for holes created inadvertently from combined sections
-        inadvertent_holes = []
+        # Determine if new holes have been created or if existing
+        # holes have been destroyed (or "filled in").
+        resultant_holes = []
         unionized_poly = unary_union([geom.geom for geom in self.geoms])
         if isinstance(unionized_poly, MultiPolygon):
             for poly in unionized_poly.geoms:
                 for interior in poly.interiors:
-                    inadvertent_holes.append(
+                    resultant_holes.append(
                         tuple(Polygon(interior).representative_point().coords[0])
                     )
 
         elif isinstance(unionized_poly, Polygon):
             if Geometry(unionized_poly).holes:
-                inadvertent_holes += Geometry(unionized_poly).holes
+                resultant_holes = Geometry(unionized_poly).holes
             else:  # Holes have been destroyed through operations
-                inadvertent_holes = []
-                self.holes = []
+                resultant_holes = []
 
-        extra_holes = []
-        if set(inadvertent_holes) - set(self.holes):
-            extra_holes = [hole for hole in inadvertent_holes if hole not in self.holes]
-
-        self.holes += extra_holes
+        self.holes = resultant_holes
 
     def calculate_perimeter(self) -> float:
         """

--- a/sectionproperties/tests/test_sections.py
+++ b/sectionproperties/tests/test_sections.py
@@ -86,8 +86,10 @@ def test__sub__():
     compound = compound + top_left
     compound = compound - top_right
     compound = compound + top_right
+    compound = compound - small_hole
 
     assert len(compound.control_points) == 3
+    assert len(compound.holes) == 1
     # Incomplete test to validate that the iterative __sub__ produces
     # three distinct regions with proper material assignments
 
@@ -167,7 +169,7 @@ def test_compound_geometry_from_points():
     assert (new_geom.geom - wkt_test_geom) == Polygon()
 
 
-def test_multinested_compound_geometry_from_points():
+def test_multi_nested_compound_geometry_from_points():
     """
     Testing a multi-nested section. This section contains three nested materials in concentric
     square rings with a hole going through the center of the whole section. This test confirms
@@ -226,7 +228,7 @@ def test_multinested_compound_geometry_from_points():
         (-31.25, 0.0),
         (-18.75, 0.0),
     ]
-    assert nested_compound.holes == [(0, 0), (0, 0), (0, 0)]
+    assert nested_compound.holes == [(0, 0)]
 
     # Section contains overlapping geometries which will result in potentially incorrect
     # plastic properties calculation (depends on user intent and geometry).

